src/components/ClusterNodes/ColumnSelector.tsx

```diff
@@ -0,0 +1,129 @@
+    setAppliedColumns: React.Dispatch<React.SetStateAction<MultiValue<columnMetadataType>>>
+}) {
+    const [selectedColumns, setSelectedColumns] = useState<MultiValue<columnMetadataType>>([])
+    const [isMenuOpen, setMenuOpen] = useState(false)
+    const [columnOptions, setColumnOptions] = useState<MultiValue<columnMetadataType>>([])
+
+    useEffect(() => {
+        setColumnOptions(columnMetadata.filter((columnData) => !columnData.isDisabled))
+        setSelectedColumns(appliedColumns)
+    }, [])
+
+    const handleApplySelectedColumns = (): void => {
+        setMenuOpen(false)
+        const _appliedColumns = [...selectedColumns].sort((a, b) => a['columnIndex'] - b['columnIndex'])
+        if (typeof Storage !== 'undefined') {
+            localStorage.appliedColumns = JSON.stringify(_appliedColumns)
+        }
+        setAppliedColumns(_appliedColumns)
+    }
+
+    const handleMenuState = (menuOpenState: boolean): void => {
+        if (menuOpenState) {
+            setSelectedColumns(appliedColumns)
+        }
+        setMenuOpen(menuOpenState)
+    }
+
+    const handleCloseFilter = (): void => {
+        handleMenuState(false)
+        setSelectedColumns(appliedColumns)
+    }
+
+    const MenuList = (props: any): JSX.Element => {
+        return (
+            <components.MenuList {...props}>
+                {props.children}
+                <div className="flex react-select__bottom bcn-0 p-8">
+                    <button className="flex cta apply-filter h-32 w-100" onClick={handleApplySelectedColumns}>
+                        Apply
+                    </button>
+                </div>
+            </components.MenuList>
+        )
+    }
+    return (
+        <ReactSelect
+            menuIsOpen={isMenuOpen}
+            name="columns"
+            value={selectedColumns}
+            options={columnOptions}
+            onChange={setSelectedColumns}
+            isMulti={true}
+            isSearchable={false}
+            closeMenuOnSelect={false}
+            hideSelectedOptions={false}
+            onMenuOpen={() => handleMenuState(true)}
+            onMenuClose={handleCloseFilter}
+            blurInputOnSelect={false}
+            autoFocus
+            components={{
+                Option: Option,
+                ValueContainer,
+                IndicatorSeparator: null,
+                ClearIndicator: null,
+                MenuList: (props) => <MenuList {...props} />,
+            }}
+            styles={{
+                ...containerImageSelectStyles,
+                menuList: (base, state) => ({
+                    ...base,
+                    borderRadius: '4px',
+                    paddingTop: 0,
+                    paddingBottom: 0,
+                }),
+                option: (base, state) => ({
+                    ...base,
+                    padding: '10px 12px',
+                    backgroundColor: state.isFocused ? 'var(--N100) !important' : 'var(--N0) !important',
```

Comment:
I think we don't need `!important` here.

Reply:
It is needed, For active state react select is changing the BG to blue